scripts/schema-publish.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Author: @ralfhandl
+
+# Run this script from the root of the repo. It is designed to be run by a GitHub workflow.
+
+for schemaDir in schemas/v3* ; do
+  vVersion=$(basename "$schemaDir")
+  version=${vVersion:1}
+  echo $version
+
+  schemas=(meta.yaml dialect.yaml schema.yaml schema-base.yaml)
+
+  # find the latest commit date for each schema
+  maxDate=""
+  declare -A datesHash
+  for schema in "${schemas[@]}"; do
+    if [ -f  "$schemaDir/$schema" ]; then
+      lastCommitDate=$(git log -1 --format="%ad" --date=short "$schemaDir/$schema")
+      if [ "$lastCommitDate" \> "$maxDate" ]; then
+        maxDate=$lastCommitDate
+      fi
+      datesHash["$schema"]=$maxDate
```

Comment:
Don't we want each `datesHash["$schema"]` to have `$schema`'s date, which would be `$lastCommitDate` and not `$maxDate`?  It's not clear to me what `$maxDate` is intended to do here.

Reply:
The schemas are visited in the order of their dependency:
* If `meta` changes, `dialect` needs to be updated. If itself has changed, the new date for `dialect` is the maximum of the two change dates.
* If `dialect` changes, `schema` needs to be updated. If itself has changed, the new date for `schema` is the maximum of the two change dates.
* If `schema` changes, `schema_base` needs to be updated. If itself has changed, the new date for `schema_base` is the maximum of the two change dates.

This is what the loop & `$maxDate` do.